woocommerce/woocommerce-paypal-payments
# Remove plugin data options.

**Issue**: #1075 

---

### Description

After the plugin uninstalling, all plugin options and other data are not removed. So after deleting and reinstalling the plugin, the onboarded state along with settings persist. Users should have a way to reset the plugin configuration back to the default state and also remove the data.

The PR will add 2 options in "Connection Tab" settings to clear the plugin data and scheduled actions when using the plugin or after uninstalling it.

### Steps to Test




1. The "**Remove PayPal Payments data from Database on uninstall**" and the "**Remove PayPal Payments data from Database(clear now button).**" exists in "Connection tab" settings.
2. When clicking on "Clear Now" button, you should first see the confirmation modal, then if clicked "ok" the data is removed, the "success" message is shown and you are redirected to "plugin connection" screen.
3. The "**Remove PayPal Payments data from Database on uninstall**" option is unchecked by default.
4. If the "**Remove PayPal Payments data from Database on uninstall**" option is checked when uninstalling the plugin then the plugin options and scheduled actions are removed

---

Closes #1075 .
